sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/SchemaRegistryAvroAsyncSerializer.java

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.avro;
+
+import com.azure.data.schemaregistry.AbstractDataSerializer;
+import com.azure.data.schemaregistry.SerializationException;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+import java.util.concurrent.Executors;
+
+/**
+ * Asynchronous registry-based serializer implementation.
+ */
+public class SchemaRegistryAvroAsyncSerializer extends AbstractDataSerializer {
+    private static final int DEFAULT_THREAD_POOL_SIZE = 8;
+
+    private final SchemaRegistryAvroSerializer serializer;
+    private final Scheduler scheduler;
+
+    /**
+     * @param serializer synchronous Avro serializer implementation
+     */
+    SchemaRegistryAvroAsyncSerializer(SchemaRegistryAvroSerializer serializer) {
+        this.serializer = serializer;
+        this.scheduler = Schedulers.fromExecutor(Executors.newFixedThreadPool(DEFAULT_THREAD_POOL_SIZE));
+    }
+
+    /**
+     * Async wrapper around sync serialization operation
+     *
+     * @param object object to be serialized to bytes
+     * @return Avro byte representation of object
+     * @throws SerializationException upon serialization operation failure
+     */
+    public Mono<byte[]> serializeAsync(Object object) throws SerializationException {
+        if (object == null) {
+            return Mono.empty();
```

Comment:
Should we use `Mono.empty()` or `Mono.just(new byte[0])`? These will have two very different things happen when dealing with Reactor streams. The former will complete the stream without an `onNext` operation, the latter will trigger an `onNext` (triggering downstream `map` and `flatMap` operators) and will contain an empty `byte[]` as expected when Avro serializes a `null`.

Reply:
Leaving this as is after discussion with Alan.

Thinking about this in terms of a user scenario makes it clear that returning empty byte array has a different meaning from returning null.  batch.tryAdd(null) throws.  batch.tryAdd(byte[0]) would not.  If there was an intermediate serialization step when doing batch.tryAdd(null), then the non-null contract breaks.

Reply:
Agree. This should return `Mono.just(new byte[0])`